pkg/kudoctl/cmd/install/install.go

```diff
@@ -83,72 +108,71 @@ func verifyFrameworks(args []string) error {
+
+	i, err := r.IndexFile.Get(name, vars.PackageVersion)
 	if err != nil {
-		return errors.Wrap(err, "sorting most recent content dir")
+		return errors.Wrapf(err, "getting %s in index file", name)
 	}
 
-	if vars.PackageVersion != "" {
-		content, err = gc.GetSpecificFrameworkContentDir(name)
-		if err != nil {
-			return errors.Wrap(err, "getting specific content dir")
+	// checking if bundle exists locally already
+	bundleName := i.Name + "-" + i.Version
+	bundlePath := fmt.Sprintf(bundlePath, vars.RepoPath, bundleName)
+
+	if _, err = os.Stat(bundlePath); err != nil && os.IsNotExist(err) {
+		if err = r.DownloadBundleFile(bundleName); err != nil {
+			return errors.Wrap(err, "failed to download bundle")
 		}
 	}
 
 	// Framework part
 
 	// Check if Framework exists
-	if !k2c.FrameworkExistsInCluster(name) {
-		err := installSingleFrameworkToCluster(name, *content.Path, gc, k2c)
-		if err != nil {
+	if !kc.FrameworkExistsInCluster(name) {
+		if err := installSingleFrameworkToCluster(name, bundlePath, kc); err != nil {
 			return errors.Wrap(err, "installing single Framework")
 		}
 	}
 
 	// FrameworkVersion part
 
 	// Get the string of the version in FrameworkVersion of a selected Framework
-	frameworkVersion, err := gc.GetFrameworkVersion(name, *content.Path)
+	frameworkVersion, err := r.GetFrameworkVersion(name, bundlePath)
 	if err != nil {
-		return errors.Wrap(err, "getting most recent FrameworkVersion version")
+		return errors.Wrap(err, "getting FrameworkVersion version")
 	}
 
 	// Check if AnyFrameworkVersion for Framework exists
-	if !k2c.AnyFrameworkVersionExistsInCluster(name) {
+	if !kc.AnyFrameworkVersionExistsInCluster(name) {
 		// FrameworkVersion CRD for Framework does not exist
-		err := installSingleFrameworkVersionToCluster(name, *content.Path, gc, k2c)
-		if err != nil {
+		if err := installSingleFrameworkVersionToCluster(name, bundlePath, kc); err != nil {
 			return errors.Wrap(err, "installing single FrameworkVersion")
 		}
 	}
 
 	// Check if FrameworkVersion is out of sync with official FrameworkVersion for this Framework
-	if !k2c.FrameworkVersionInClusterOutOfSync(name, frameworkVersion) {
+	if !kc.FrameworkVersionInClusterOutOfSync(name, frameworkVersion) {
 		// This happens when the given FrameworkVersion is not existing. E.g.
 		// when a version has been installed that is not part of the official kudobuilder/frameworks repo.
 		if !vars.AutoApprove {
 			fmt.Printf("No official FrameworkVersion has been found for \"%s\". "+
 				"Do you want to install one? (Yes/no) ", name)
 			if helpers.AskForConfirmation() {
-				err := installSingleFrameworkVersionToCluster(name, *content.Path, gc, k2c)
-				if err != nil {
-					return errors.Wrap(err, "installing single FrameworkVersion")
+				if err := installSingleFrameworkVersionToCluster(name, bundlePath, kc); err != nil {
+					return errors.Wrapf(err, "installing single FrameworkVersion %s", name)
```

Comment:
What about just: "installing Framework version %s"?

Reply:
I've changed it to `installing FrameworkVersion CRD for framework`